pkg/conference/participant.go

```diff
@@ -0,0 +1,51 @@
+package conference
+
+import (
+	"encoding/json"
+
+	"github.com/matrix-org/waterfall/pkg/peer"
+	"github.com/matrix-org/waterfall/pkg/signaling"
+	"github.com/pion/webrtc/v3"
+	"github.com/sirupsen/logrus"
+	"maunium.net/go/mautrix/event"
+	"maunium.net/go/mautrix/id"
+)
+
+// Things that we assume as identifiers for the participants in the call.
+// There could be no 2 participants in the room with identical IDs.
+type ParticipantID struct {
+	UserID   id.UserID
+	DeviceID id.DeviceID
+	CallID   string
+}
+
+// Participant represents a participant in the conference.
+type Participant struct {
+	id              ParticipantID
+	logger          *logrus.Entry
+	peer            *peer.Peer[ParticipantID]
+	remoteSessionID id.SessionID
+	streamMetadata  event.CallSDPStreamMetadata
+	publishedTracks map[event.SFUTrackDescription]*webrtc.TrackLocalStaticRTP
```

Comment:
Could we not use `event.SFUTrackDescription` here, please? It doesn't feel good that the Matrix event types and internal types being coupled if they aren't exactly the same thing. Also, `event.SFUTrackDescription` optionally  includes resolution, so it just feels wrong to use it as a key here since we'll never want to use resolution in the key

Reply:
In general, I totally agree with it. I think the only reason I used it like this is that it was not clear what exactly we are going to use [as a track identifier](https://github.com/matrix-org/matrix-spec-proposals/pull/3898#discussion_r1038568858), but I agree that having a separate identifier for tracks makes more sense (it would also simplify the adjacent functions that compare things).

Reply:
Yep, let's merge this ASAP - have made https://github.com/matrix-org/waterfall/issues/56. I might actually have a look into that myself if I get a change before you